internal/controller/controller.go

```diff
@@ -234,15 +242,56 @@ func (c *httpController) exportRecordedData(writer http.ResponseWriter, request
 		return
 	}
 
-	jsonResponse, err := json.Marshal(recordedData)
-	if err != nil {
+	if reflect.DeepEqual(recordedData, &dtos.RecordedData{}) {
+		writer.WriteHeader(http.StatusNoContent)
+		_, _ = writer.Write([]byte("no recorded data found"))
+		return
+	}
+
+	compression := request.URL.Query().Get("Content-Encoding")
+	switch compression {
+	case noCompression:
+		jsonResponse, err := json.Marshal(recordedData)
+		if err != nil {
+			writer.WriteHeader(http.StatusInternalServerError)
+			_, _ = writer.Write([]byte("failed to marshal recorded data"))
+			return
+		}
+		writer.WriteHeader(http.StatusOK)
```

Comment:
I think we should be setting `"Content-Type", "application/json"` on all our JSON responses.